frontend/packages/dev-console/src/components/project-access/project-access-form-submit-utils.ts

```diff
@@ -86,16 +86,30 @@ export const getRemovedRoles = (
 export const sendRoleBindingRequest = (
   verb: string,
   roles: UserRoleBinding[],
-  roleBinding: RoleBinding,
+  namespace: string,
 ) => {
-  const rb = _.clone(roleBinding);
-  const finalArray = [];
+  const roleBinding: RoleBinding = {
+    apiVersion: 'rbac.authorization.k8s.io/v1',
+    kind: 'RoleBinding',
+    metadata: {
+      name: '',
+      namespace,
+    },
+    roleRef: {
+      apiGroup: 'rbac.authorization.k8s.io',
+      kind: 'ClusterRole',
+      name: '',
+    },
+    subjects: [],
+  };
```

Comment:
This might work (as before) because the network call serializes the body (roleBinding) in `sendK8sRequest`

But I would recommend to not modifying the roleBinding in every `_.forEach(roles,...` loops and passing the same instance (===) multiple times to another API call.

Can you move this `const roleBinding: Rolebinding...` into the loop? And then we might also can set the metadata.name and maybe other fields when constructing it instead of assign/modify it.

```suggestion
  const roleBinding: RoleBinding = {
    apiVersion: 'rbac.authorization.k8s.io/v1',
    kind: 'RoleBinding',
    metadata: {
      name: roleBindingName,
      namespace,
    },
    roleRef: {
      apiGroup: 'rbac.authorization.k8s.io',
      kind: 'ClusterRole',
      name: user.role,
    },
    subjects,
  };
```